File: Data Structures & Algorithms/products-of-array-discluding-self/submission-14.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> store(nums.size(), 1);
        for(int i = 1; i < nums.size(); i++){
            store[i] = nums[i-1] * store[i-1];
        }
        int prefix = 1;
        for(int i = nums.size()-1; i >= 0; i--){
            store[i] *= prefix;
            prefix *= nums[i];
        }
        return store;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-14.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        curr = []
        nums.sort()
        def dfs(i, tar):
            if tar == 0:
                res.append(curr.copy())
                return
            for j in range(i, len(nums)):
                if tar < nums[j]:
                    return 
                curr.append(nums[j])
                dfs(j, tar - nums[j])
                curr.pop()
        dfs(0, target)
        return res